### DISCUSSION TITLE
Reset FSM within react-context

### DISCUSSION THREAD

**Discussion by johannes-darms** [statelyai/xstate#3338] [Q&A]
I create a FSM inside a React context with the call "useInterpret(exampleFSM)"[1]. Is there any way to reset or restart the FSM, i.e. clear everything (context, history and return to the initial state)?

In my use case, an FSM reaches a terminal state and the process is considered complete. The process could be executed one more time, so I want to reset/re-create the FSM so that another input can be consumed.  Of course, I could model the FSM without the terminal node and add a loop to the initial node. However, since no information about the previous execution is needed, I would rather start with a new instance.

[1] https://xstate.js.org/docs/recipes/react.html#global-state-react-context

**Comment by davidkpiano** [comment]
Since it's tied to the component, you can "unmount/remount" the component (basically a new instance) by assigning it a new `key={...}`. Would that help?

**Reply by johannes-darms** [reply]
Thanks for the fast reply. Sure that would work, but similar to remodeling the FSM it feels a bit cumbersome. I hoped I missed a bit of documentation and their is some kind of `reset` function exposed via the hook or Interpreter.
